digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -0,0 +1,1739 @@
+                files.add(ImmutableStagingFile.builder()
+                        .template(template)
+                        .file(remoteFile)
+                        .build());
+            }
+
+            return remoteFile;
+        }
+
+        void stageFiles()
+        {
+            if (files.isEmpty()) {
+                return;
+            }
+
+            TransferManager transferManager = new TransferManager(s3);
+            List<PutObjectRequest> requests = new ArrayList<>();
+
+            for (StagingFile f : files) {
+                logger.info("Staging {} -> {}", f.file().reference().filename(), f.file().s3Uri());
+                requests.add(stagingFilePutRequest(f));
+            }
+
+            try {
+                List<Upload> uploads = requests.stream()
+                        .map(transferManager::upload)
+                        .collect(toList());
+
+                for (Upload upload : uploads) {
+                    try {
+                        upload.waitForCompletion();
+                    }
+                    catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+                    }
+                }
+            }
+            finally {
+                transferManager.shutdownNow(false);
+                requests.forEach(r -> closeQuietly(r.getInputStream()));
+            }
+        }
+
+        private PutObjectRequest stagingFilePutRequest(StagingFile file)
+        {
+            AmazonS3URI uri = file.file().s3Uri();
+            FileReference reference = file.file().reference();
+
+            switch (reference.type()) {
+                case LOCAL: {
+                    if (file.template()) {
+                        try {
+                            String content = workspace.templateFile(templateEngine, reference.filename(), UTF_8, params);
+                            byte[] bytes = content.getBytes(UTF_8);
+                            ObjectMetadata metadata = new ObjectMetadata();
+                            metadata.setContentLength(bytes.length);
+                            return new PutObjectRequest(uri.getBucket(), uri.getKey(), new ByteArrayInputStream(bytes), metadata);
+                        }
+                        catch (IOException | TemplateException e) {
+                            throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+                        }
+                    }
+                    else {
+                        return new PutObjectRequest(uri.getBucket(), uri.getKey(), workspace.getFile(reference.filename()));
+                    }
+                }
+                case RESOURCE: {
+                    byte[] bytes;
+                    try {
+                        bytes = Resources.toByteArray(new URL(reference.reference().get()));
```

Comment:
[trivial] Calling `reference.reference().get()` without `referene.isPresent()` depends on the implementation that the types except for `DIRECT` have `Optional.of(reference)`. Maybe we can refactor `RemoteFile` later with class polymorphism.

Like this?
```
abstract class RemoteFile
{
    private final AmazonS3URI uri;

    protected RemoteFile(AmazonS3URI uri)
    {
        this.uri = uri;
    }

    public AmazonS3URI getUri()
    {
        return uri;
    }

    public abstract PutObjectRequest createUploadRequest();
}

class RemoteFileWithReference
    extends RemoteFile
{
    private final FileReference reference;

    public RemoteFileWithReference(AmazonS3URI uri, FileReference reference)
    {
        super(uri);
        this.reference = reference;
    }
}

class RemoteFileWithLocalFile
    extends RemoteFileWithReference
{
    public PutObjectRequest createUploadRequest()
    {
        :
    }
}

class RemoteFileWithDirectUpload
    extends RemoteFile
{
    private final byte[] contents;

    public RemoteFileWithDirectUpload(AmazonS3URI uri, byte[] contents)
    {
        super(uri);
        this.contents = contents;
    }

    public PutObjectRequest createUploadRequest()
    {
        :
    }
}
```